hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -447,6 +447,25 @@ protected HoodieBaseFile addBootstrapBaseFileIfPresent(HoodieFileGroupId fileGro
     }
   }
 
+  @Override
+  public final Stream<HoodieBaseFile> getLatestBaseFilesBeforeOrOn(String maxCommitTime) {
+    try {
+      readLock.lock();
+      return fetchAllStoredFileGroups()
```

Comment:
I see an opportunity for code re-use between this and getLatestBaseFilesBeforeOrOn(String partitionStr, String maxCommitTime)(lines 470 to 486).

Reply:
Infact we could change the signature of existing method to 
```
getLatestBaseFilesBeforeOrOn(Option<String> partitionStr, String maxCommitTime)
```
and not introduce a new method.